packages/react-composites/src/composites/ChatComposite/file-sharing/FileDownloadHandler.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { FileMetadata } from './FileUpload';
+
+/**
+ * @beta
+ *
+ * A callback function for handling file downloads.
+ * The function needs to return a promise that resolves to a file download URL.
+ * If the promise is rejected, the {@link Error.message} will be used to display an error message to the user.
+ *
+ * Sample Usage:
+ * ```ts
+ * const fileDownloadHandler: FileDownloadHandler = async (userId, fileData) => {
+ *   if (isUnauthorizedUser(userId)) {
+ *     throw new Error('You don’t have permission to download this file.');
```

Comment:
This sample code makes me think that `FileDownloadHandler` should have a way of notifying us that download failed, so that we can show the appropriate message in the UI (instead of throwing an exception).

Further, we could provide a small set of error causes that can be returned 

`const fileDownloadFailureReason = 'unauthorized' | 'notFound'` etc.

Reply:
Yes, the current idea is to catch the thrown error via `fileDownloadHandler` and display this error message.

Reply:
What do you think about that?

Reply:
If contoso if throwing the error and providing an error message, why do we need to care about localization of that message?
This way allows contoso to display any error message not limited by `fileDownloadFailureReason`

Reply:
I was thinking of showing error messages on the UI (to the end user). We do not show any strings on the UI that are not localized.

For any strings that go on our UI, one of two paths must be followed:
* We own the string -- it is localized
* Contoso owns the string -- we provide a way for Contoso to inject this UI component.

We currently have way (and no plans) for Contoso to inject error bars or similar UX surface, so I was thinking of us showing an error string in the UI (probably close to the FileCard, saying we failed to upload, and why).

Reply:
Talked IRL. We're showing strings that Contoso gives us. They need to localize if they want.